### PR TITLE
taking care of the last moves in initial sequence

### PR DESCRIPTION
**Pull request by yohm**
(Currently, work in progress, but this is a working code. More refactoring and testing are needed.)

I revised the `infer_best_response_and_expected_payoffs` function such that it takes the last move in the initial sequence.

First, we calculate the `posterior_distribution` to infer $p_CC,p_CD,p_DC,p_DD$. Then, we use this posterior distribution in `infer_best_response_and_expected_payoffs` function.

### CODE REVIEW

**Comment by Nikoleta-v3** [comment]
Thank you for working on this @yohm 😄 I guess one thing I am not convinced of is the `posterior_distribution`. You are still working on that function?

**Comment by yohm** [comment]
`posterior_distribution` is already complete.
In this function, the posterior distributions are represented by an array of length 16, ignoring the initial move.
We check every transition. If the transition is inconsistent with the strategy, I set the posterior distribution for that strategy to zero, as it cannot happen. Finally, we normalize the distribution.

I assumed that there is no error and the strategies are pure memory-1.

**Comment by yohm** [comment]
@Nikoleta-v3 (Minimal) refactoring & unit testing are done. It would be easier to look at `test_mockup.py` first.

File: src/mockup.py [hunk 1]
```diff
@@ -156,8 +208,9 @@ def long_term_payoffs(
 
             # inferring co-player and best response
             bs, exp_p = infer_best_response_and_expected_payoffs(history, payoff_matrix)
-            lt_payoffs = long_term_payoffs(
-                opening_payoffs, exp_p, delta
+            # exclude the last payoff because it is included in exp_p
+            lt_payoffs = long_term_total_payoff(
+                opening_payoffs[:-1], exp_p, delta
```
Comment (Nikoleta-v3) [inline src/mockup.py:213]: Niiiice 👍🏻

**Review by Nikoleta-v3 — APPROVED** [review]
You are right @yohm I was mis-understanding your function.

Thank you for the tests 👍🏻

**Comment by Nikoleta-v3** [comment]
Can I merge this?

**Pull request merged**